gammapy/modeling/models/temporal.py

```diff
@@ -334,6 +334,74 @@ def integral(self, t_min, t_max, **kwargs):
         return integral / self.time_sum(t_min, t_max)
 
 
+class GeneralizedGaussianTemporalModel(TemporalModel):
+    r"""A generalized Gaussian temporal profile
+
+    ..math::
+            F(t) = exp( - 0.5 * (\frac{|t - t_{ref}|}{t_rise}) ^ {1 / \eta})   for  t < t_ref
+            
+            F(t) = exp( - 0.5 * (\frac{|t - t_{ref}|}{t_decay}) ^ {1 / \eta})   for  t > t_ref            
+
+    Parameters
+    ----------
+    t_ref: `~astropy.units.Quantity`
+        The time of the pulse's maximum intensity.
+    t_rise : `~astropy.units.Quantity`
+        Rise time constant.
+    t_decay : `~astropy.units.Quantity`
+        Decay time constant.
+    eta : `~astropy.units.Quantity`
+        Inverse pulse sharpness -> higher values implies a more peaked pulse
+    
+    """
+
+    tag = ["GeneralizedGaussianTemporalModel", "gengauss"]
+
+    _t_ref_default = Time("2000-01-01")
+    t_ref = Parameter("t_ref", _t_ref_default.mjd, unit = "day", frozen=False)
+    t_rise = Parameter("t_rise", "1d", frozen=False)
+    t_decay = Parameter("t_decay", "1d", frozen=False)
+    eta = Parameter("eta", 1/2, unit = "", frozen=False)
+
+    @staticmethod
+    def evaluate(time, t_ref, t_rise, t_decay, eta):
+        val_rise = np.exp( - 0.5 * (np.abs((time - t_ref).value) ** (1/eta)) / (t_rise.to_value("d") ** (1/eta)))
+        val_decay = np.exp( - 0.5 * (np.abs((time - t_ref).value) ** (1/eta)) / (t_decay.to_value("d") ** (1/eta)))
+        val = np.where(time < t_ref, val_rise, val_decay)
+        return val
+    
+    def integral(self, t_min, t_max, **kwargs):
+        """Evaluate the integrated flux within the given time intervals
+
+        Parameters
+        ----------
+        t_min: `~astropy.time.Time`
+            Start times of observation
+        t_max: `~astropy.time.Time`
+            Stop times of observation
+
+        Returns
+        -------
+        norm : float
+            Integrated flux norm on the given time intervals
+        """
+        
+        pars = self.parameters
+        t_rise = pars["t_rise"].quantity.to_value("day")
+        t_decay = pars["t_decay"].quantity.to_value("day")
+        eta = pars["eta"].quantity
+        t_ref = Time(pars["t_ref"].quantity, format = "mjd")
+                     
+        def gen_gauss(time,t_ref, t_rise, t_decay, eta):
```

Comment:
Why is this re-declaration needed? I think directly calling `.evaluate()` shock work fine...

Reply:
Would great if you had time to address this remaining comment @aaguasca...Thanks!

Reply:
I was waiting for an answer in issue #3736 before uploading any commit. Depending on the answer, I should change the code, as I cannot introduce more than one interval of time in quad.